lib/teleterm/daemon/daemon_test.go

```diff
@@ -90,14 +90,7 @@ func (m *mockGatewayCreator) CreateGateway(ctx context.Context, params clusters.
 
 	ca := gatewaytest.MustGenCACert(m.t)
 
-	if params.TargetURI.IsDB() {
-		keyPairPaths := gatewaytest.MustGenAndSaveCert(m.t, ca, identity)
-
-		config.CertPath = keyPairPaths.CertPath
-		config.KeyPath = keyPairPaths.KeyPath
-	}
-
-	if params.TargetURI.IsKube() {
+	if params.TargetURI.IsKube() || params.TargetURI.IsDB() {
```

Comment:
This conditional can be removed.

<details>
<summary>Patch</summary>

```diff
diff --git a/lib/teleterm/daemon/daemon_test.go b/lib/teleterm/daemon/daemon_test.go
index 0d97c0874b..7617fc4653 100644
--- a/lib/teleterm/daemon/daemon_test.go
+++ b/lib/teleterm/daemon/daemon_test.go
@@ -64,19 +64,7 @@ func (m *mockGatewayCreator) CreateGateway(ctx context.Context, params clusters.
 		hs.Close()
 	})
 
-	config := gateway.Config{
-		LocalPort:             params.LocalPort,
-		TargetURI:             params.TargetURI,
-		TargetUser:            params.TargetUser,
-		TargetName:            params.TargetURI.GetDbName() + params.TargetURI.GetKubeName(),
-		TargetSubresourceName: params.TargetSubresourceName,
-		Protocol:              defaults.ProtocolPostgres,
-		Insecure:              true,
-		WebProxyAddr:          hs.Listener.Addr().String(),
-		TCPPortAllocator:      m.tcpPortAllocator,
-		KubeconfigsDir:        m.t.TempDir(),
-	}
-
+	ca := gatewaytest.MustGenCACert(m.t)
 	identity := tlsca.Identity{
 		Username: "user",
 		Groups:   []string{"test-group"},
@@ -88,11 +76,18 @@ func (m *mockGatewayCreator) CreateGateway(ctx context.Context, params clusters.
 		KubernetesCluster: params.TargetURI.GetKubeName(),
 	}
 
-	ca := gatewaytest.MustGenCACert(m.t)
-
-	if params.TargetURI.IsKube() || params.TargetURI.IsDB() {
-		cert := gatewaytest.MustGenCertSignedWithCA(m.t, ca, identity)
-		config.Cert = cert
+	config := gateway.Config{
+		LocalPort:             params.LocalPort,
+		TargetURI:             params.TargetURI,
+		TargetUser:            params.TargetUser,
+		TargetName:            params.TargetURI.GetDbName() + params.TargetURI.GetKubeName(),
+		TargetSubresourceName: params.TargetSubresourceName,
+		Protocol:              defaults.ProtocolPostgres,
+		Insecure:              true,
+		WebProxyAddr:          hs.Listener.Addr().String(),
+		TCPPortAllocator:      m.tcpPortAllocator,
+		KubeconfigsDir:        m.t.TempDir(),
+		Cert:                  gatewaytest.MustGenCertSignedWithCA(m.t, ca, identity),
 	}
 
 	gateway, err := gateway.New(config)

```
</details>